src/lib/diagnostics.ts

```diff
@@ -6,211 +6,185 @@
+    addDiagnostics(diags: CodeAnalyzerDiagnostic[]): void
     clearAllDiagnostics(): void
-    clearDiagnostic(uri: vscode.Uri, diag: vscode.Diagnostic);
+    clearDiagnostic(diag: CodeAnalyzerDiagnostic): void
     clearDiagnosticsInRange(uri: vscode.Uri, range: vscode.Range): void
-    clearDiagnosticsForSelectedFiles(selections: vscode.Uri[], commandName: string): Promise<void>
-    displayAsDiagnostics(allTargets: string[], convertibles: DiagnosticConvertible[]): void
+    clearDiagnosticsForFiles(uris: vscode.Uri[]): void
 }
 
 export class DiagnosticManagerImpl implements DiagnosticManager {
     private readonly diagnosticCollection: vscode.DiagnosticCollection;
-    private readonly telemetryService: TelemetryService;
-    private readonly logger: Logger;
 
-    public constructor(diagnosticCollection: vscode.DiagnosticCollection, telemetryService: TelemetryService, logger: Logger) {
+    public constructor(diagnosticCollection: vscode.DiagnosticCollection) {
         this.diagnosticCollection = diagnosticCollection;
-        this.telemetryService = telemetryService;
-        this.logger = logger;
+    }
+
+    public addDiagnostics(diags: CodeAnalyzerDiagnostic[]) {
+        const uriToDiagsMap: Map<vscode.Uri, CodeAnalyzerDiagnostic[]> = groupByUri(diags);
+        for (const [uri, diags] of uriToDiagsMap) {
+            this.addDiagnosticsForUri(uri, diags);
+        }
     }
 
     public clearAllDiagnostics(): void {
         this.diagnosticCollection.clear();
     }
 
-    public clearDiagnostic(uri: vscode.Uri, diagnostic: vscode.Diagnostic): void {
-        const currentDiagnostics: readonly vscode.Diagnostic[] = this.diagnosticCollection.get(uri) || [];
+    public clearDiagnostic(diagnostic: CodeAnalyzerDiagnostic): void {
+        const uri: vscode.Uri = diagnostic.uri;
+        const currentDiagnostics: readonly vscode.Diagnostic[] = this.getDiagnosticsForUri(uri);
         const updatedDiagnostics: vscode.Diagnostic[] = currentDiagnostics.filter(diag => diag !== diagnostic);
-        this.diagnosticCollection.set(uri, updatedDiagnostics);
+        this.setDiagnosticsForUri(uri, updatedDiagnostics);
     }
 
     public dispose(): void {
         this.clearAllDiagnostics();
     }
 
-    /**
-     * Clear diagnostics for a specific files
-     */
-    async clearDiagnosticsForSelectedFiles(selections: vscode.Uri[], commandName: string): Promise<void> {
-        const startTime = Date.now();
-
-        try {
-            const targets: string[] = await targeting.getTargets(selections);
-
-            for (const target of targets) {
-                this.diagnosticCollection.delete(vscode.Uri.file(target));
-            }
-
-            // TODO: It doesn't make sense that we are performing telemetry here in my opinion. This should be
-            //        moved to an associated action.
-            this.telemetryService.sendCommandEvent(Constants.TELEM_SUCCESSFUL_STATIC_ANALYSIS, {
```

Comment:
We discussed in review that this didn't make sense here, and might in fact have been affecting telemetry data in an unhelpful way, so good to clean up while here 👍